device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java

```diff
@@ -193,7 +197,41 @@ public static ModuleClient createFromEnvironment(IotHubClientProtocol protocol,
         {
             log.debug("Creating module client with the provided connection string");
 
-            //Codes_SRS_MODULECLIENT_34_020: [If an edgehub or iothub connection string is present, this function shall create a module client instance using that connection string and the provided protocol.]
+            //Check for a different default cert to be used
+            String alternativeDefaultTrustedCert = envVariables.get(EdgeCaCertificateFileVariableName);
+            SSLContext sslContext;
+            if (alternativeDefaultTrustedCert != null && !alternativeDefaultTrustedCert.isEmpty())
+            {
+                log.debug("Configuring module client to use the configured alternative trusted certificate");
+                try
+                {
+                    sslContext = IotHubSSLContext.getSSLContextFromFile(alternativeDefaultTrustedCert);
+                }
+                catch (CertificateException | IOException | KeyStoreException | NoSuchAlgorithmException | KeyManagementException e)
+                {
+                    throw new ModuleClientException("Failed to create an SSLContext instance from the provided trusted cert file path", e);
+                }
+            }
+            else
+            {
+                sslContext = new IotHubSSLContext().getSSLContext();
```

Comment:
And if that public certificate isn't present, then just create a default SSLContext instead